thorlcr/activities/nsplitter/thnsplitterslave.cpp

```diff
@@ -163,7 +163,7 @@ class NSplitterSlaveActivity : public CSlaveActivity, implements ISharedSmartBuf
         IMPLEMENT_IINTERFACE_USING(CSimpleInterface);
 
         CDelayedInput(NSplitterSlaveActivity &_activity) : CThorDataLink(&_activity), activity(&_activity), id(0) { }
-        void setInput(CSplitterOutputBase *_input, unsigned _id=0)
+        virtual void setInputSplitterOutputBase(CSplitterOutputBase *_input, unsigned _id=0)
```

Comment:
Ok, I sort of understand why the change was made now.

Splitter doesn't know how many outputs will be connected up upfront.
It creates pseudo outputs (CDelayInput objects with are IThorDataLink's), where the input can be connected at splitter start() time.

At splitter start time, the existing code assumes the outputs are still instances of what it added (because they were guaranteed to be before) and casts them to CDelayedInput, so as to call the class member setInput().

The new member added to IThorDataLink in this PR (setInputSplitterOutputBase()) , removes the need for casting by introducing it into the common IThorDataLink interface, where all but splitter CDelayedInput implementation do nothing.

However this member isn't something activities in general want to use/know about, so it doesn't belong in IThorDataLink.

I think you achieve the same as the pre-existing code, by tracking the CDelayedInput objects created in the splitter code separately (instead of using 'outputs') and use them directly instead.